reactor-netty-http/src/main/java/reactor/netty/http/server/HttpTrafficHandler.java

```diff
@@ -507,10 +509,19 @@ protected void channelIdle(ChannelHandlerContext ctx, IdleStateEvent evt) {
 		static void addIdleTimeoutHandler(ChannelPipeline pipeline, @Nullable Duration idleTimeout) {
 			if (idleTimeout != null) {
 				String baseName = pipeline.get(NettyPipeline.HttpCodec) != null
-						? NettyPipeline.HttpCodec : NettyPipeline.H2CUpgradeHandler;
+						? NettyPipeline.HttpCodec : pipeline.get(NettyPipeline.H2CUpgradeHandler) != null
+						? NettyPipeline.H2CUpgradeHandler : null;
+
+				if (baseName == null) {
+					ChannelHandler httpServerCodec = pipeline.get(HttpServerCodec.class);
+					if (httpServerCodec != null) {
+						baseName = pipeline.context(httpServerCodec).name();
+					}
+				}
+
 				pipeline.addBefore(baseName,
-				                   NettyPipeline.IdleTimeoutHandler,
-				                   new IdleTimeoutHandler(idleTimeout.toMillis()));
+						NettyPipeline.IdleTimeoutHandler,
```

Comment:
This change is formatting and we can remove it.

Reply:
Sure!

Reply:
Done, reverted this change